lib/resolveTaskFn.js

```diff
@@ -159,7 +159,12 @@ export const resolveTaskFn = ({
 
   return async (ctx = getInitialState()) => {
     const execaChildProcess = shell
-      ? execaCommand(isFn ? command : `${command} ${files.join(' ')}`, execaOptions)
+      ? execaCommand(
+          isFn
+            ? command
+            : `${command} ${files.map((file) => file.replaceAll(' ', '\\ ')).join(' ')}`,
+          execaOptions
+        )
```

Comment:
I see this only targets the `shell` option, so that's why most tests pass. Can you extract this to some named function like:

```javascript
/**
 * @see https://github.com/sindresorhus/execa/blob/f4b8b3ab601c94d1503f1010822952758dcc6350/lib/command.js#L32-L37
 */
const escapeSpaces = (input) => input.replaceAll(' ', '\\ ')
```

and then use it here:

```suggestion
      ? execaCommand(
          isFn
            ? command
            : `${command} ${files.map(escapeSpaces).join(' ')}`,
          execaOptions
        )
```